### DISCUSSION TITLE
Signed Certificates

### DISCUSSION THREAD

**Discussion by Epicism** [telekom-security/tpotce#1376] [Q&A]
Hello,

I want to use let's encrypt to create signed certificates on honeypots. Is there an easy way to do this?

**Comment by jcilley-mhvfcu** [comment]
This would be a fantastic feature to include into a future release should the dev team feel like they can build it in.

**Reply by t3chn0m4g3** [reply]
The challenge with Let's Encrypt is, that it actively tries to connect to `tcp/80` and `tcp/443` in order to issue the certificate and usually runs in an error if the ports do not answer as expected. Building a solution that just works for everyone would need a different approach, i.e. by using [Caddy](https://caddyserver.com/) as a reverse proxy for the honeypots in question. Currently there is no focus on this, might be revisited after the next release. It is probably more feasible to add a white paper how to adjust T-Pot for Caddy with LE certs and reverse proxying compatible honeypots.

**Reply by jcilley-mhvfcu** [reply]
Not a problem at all. It is not an absolutely necessary feature for me by any means. More a thought for something that might be able to be done down the road when someone has time. T-Pot is an amazing tool, and now that I am running a network of three sensors, it is amazing to shine a light on what kinds of threat actors there are out there.
